### ISSUE TITLE
Add public interface for transferring TPC.

### ISSUE THREAD

**Issue by rantan** [chaintope/glueby#18]
Steps

* Design public interface and review on this issue.
* Impl the interface and create PR

**Comment by fryota142** [comment]
# Supported Features
  - Transfer TPC between Tapyrus Core wallets

## Add file and method
### ● add file ( lib/glueby/contract/tpc.rb )
### ● Glueby::Contract::Tpc.transfer(sender_wallet:, receiver_wallet:, amount:, script: nil, fee_provider: FixedFeeProvider.new)
・Argument description
 - sender_wallet : set sender's wallet
 - receiver_wallet : set receiver's wallet
 - amount : set transfer amount(tapyrus)
 - script : set ScriptPubkey if you use script [default: nil]
 - fee_provider : set transaction fee(satoshi) [default: FixedFeeProvider.new]

# Code Examle
```ruby: sample.rb
# load or create wallet
>> sender_wallet = Glueby::Internal::Wallet.load("611e05e8b8d003763b4c5e789cc56d2d")
>> receiver_wallet = Glueby::Internal::Wallet.create
                                  or
                     Glueby::Internal::Wallet.load("e202fb030602e3079cb51810ed37c498")

# transfer TPC and return txid
>> Glueby::Contract::Tpc.transfer(sender_wallet: sender_wallet, receiver_wallet: receiver_wallet, amount: 100000, script: nil, fee_provider: FixedFeeProvider.new)
=> "14d430605d5c907efee68a9b28429c8e9a50b90df71e80d0dbde79959386d69e"
```

**Comment by azuchi** [comment]
`script` argument dose not necessary.

`TPC` is currency unit, therefore I think 
* `Payment` is appropriate in the context of contracts. 
* `Payment` is preferably an AR object like `Glueby::Contract::AR::Timestamp`.

@rantan @Yamaguchi What do you think?

**Comment by Yamaguchi** [comment]
sender_wallet and receiver_wallet should be "Glueby::Wallet" rather than "Glueby::Internal::Wallet".
We should not use internal classes and modules directly in client application.

**Comment by Yamaguchi** [comment]
> `script` argument dose not necessary.

👍 

> `TPC` is currency unit, therefore I think
> 
> * `Payment` is appropriate in the context of contracts.

👍 
And if you change the class name, the method name had better to be considered again.
`Payment.transfer(...)` or `Payment.create(...)` or any other?

> * `Payment` is preferably an AR object like `Glueby::Contract::AR::Timestamp`.

👍 . But I think this is the next step after merging first implementation.

**Comment by fryota142** [comment]
Thank you for reviewing.

> script argument dose not necessary.

OK. I delete this argument.

>TPC is currency unit, therefore I think
> - Payment is appropriate in the context of contracts.
>>And if you change the class name, the method name had better to be considered again.
>>Payment.transfer(...) or Payment.create(...) or any other?

I rename from `tpc` to `payment` file name and class name.
And I change method name to `Payment.transfer(...)`

> sender_wallet and receiver_wallet should be "Glueby::Wallet" rather than "Glueby::Internal::Wallet".

I didn't understand internal classes and modules.
I change to use `Glueby::Wallet`.

**Comment by rantan** [comment]
This feature is merged at #25
